plugins/BEdita/Core/src/Model/Entity/ObjectEntity.php

```diff
@@ -185,6 +185,33 @@ protected function getRelationships()
                 ],
                 true
             );
+            $entityDest = $this->getTable()->associations()->getByProperty($relationship)->getTarget()->newEntity();
+            if ($entityDest instanceof JsonApiSerializable) {
+                $destObj = $entityDest->jsonApiSerialize(JsonApiSerializable::JSONAPIOPT_BASIC);
+                if (empty($destObj['type'])) {
+                    $options = ['name' => $relationship];
+                    $result = TableRegistry::get('ObjectTypes')
```

Comment:
There is. Object types cache already includes information about relations and relatable object types. You might take a look at `RelationsBehavior` to see how it's done in there, or I might take care of this myself in the afternoon.